Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkBasic/PsychEyelinkDispatchCallback.m

```diff
@@ -129,326 +161,419 @@
     drawInstructions=0;
     EyelinkDrawCalibrationTarget(eyewin, el, calxy);
+    if strcmpi(el.calTargetType, 'video')
+        return;
+    end
 end
 
-% Need to draw instructions?
-if drawInstructions==1
-    EyelinkDrawInstructions(eyewin, el,msg);
-    drawInstructions=0;
+if drawInstructions == 1  % Draw Instructions
+    EyelinkDrawInstructions(eyewin, el, msg);
+    drawInstructions = 0;
 end
 
-% Show it: We disable synchronization of Matlab to the vertical retrace.
-% This way, display update itself is still synced and tear-free, but we
-% don't waste time waiting for swap completion. Potentially higher
-% performance for calibration displays and eye camera image updates...
-% Neither do we erase buffer
-Screen('Flip', eyewin, [], 1, 1);
+dontsync = 1;
+dontclear = 0;
+Screen('Flip', eyewin, [], dontclear, dontsync);   % Show it
 
-% Some counter, just to measure update rate:
-drawcount = drawcount + 1;
-
-% Done. Return from callback:
 return;
 
 
-function EyelinkDrawInstructions(eyewin, el,msg)
-oldFont=Screen(eyewin,'TextFont',el.msgfont);
-oldFontSize=Screen(eyewin,'TextSize',el.msgfontsize);
-DrawFormattedText(eyewin, el.helptext, 20, 20, el.msgfontcolour, [], [], [], 1);
 
-if el.displayCalResults && ~isempty(msg)
-    DrawFormattedText(eyewin, msg, 20, 150, el.msgfontcolour, [], [], [], 1);
-end
+% Start of nested EyelinkDraw* function declarations
 
-Screen(eyewin,'TextFont',oldFont);
-Screen(eyewin,'TextSize',oldFontSize);
-
-function  imgtitle=EyelinkDrawCameraImage(eyewin, el, eyelinktex, imgtitle,newImage)
-persistent lasttitle;
-global dh dw offscreen;
-try
-    if ~isempty(eyelinktex)
-        eyerect=Screen('Rect', eyelinktex);
-        % we could cash some of the below values....
-        wrect=Screen('Rect', eyewin);
-        [width, heigth]=Screen('WindowSize', eyewin);
-        dw=round(el.eyeimgsize/100*width);
-        dh=round(dw * eyerect(4)/eyerect(3));
+    function EyelinkDrawClearScreen(eyewin, el)
+        if el.winInfo.StereoMode ~= 0 
+            drawScreens = 2; % stereoscopic drawing
+        else
+            drawScreens = 1; % non-stereoscopic drawing
+        end
+        for it = 0:drawScreens-1 
+            Screen('SelectStereoDrawBuffer', eyewin, it); % select left eye window
+            Screen('FillRect', eyewin, el.backgroundcolour);
+        end
+    end
 
-        drect=[ 0 0 dw dh ];
-        drect=CenterRect(drect, wrect);
-        Screen('DrawTexture', eyewin, eyelinktex, [], drect);
-        %     fprintf('EyelinkDrawCameraImage:DrawTexture \n');
+    function EyelinkDrawInstructions(eyewin, el,msg)
+        if el.winInfo.StereoMode ~= 0 
+            drawScreens = 2; % stereoscopic drawing
+        else
+            drawScreens = 1; % non-stereoscopic drawing
+        end
+        
+        for it = 0:drawScreens-1 
+            Screen('SelectStereoDrawBuffer', eyewin, it); % select left eye window
+            oldFont=Screen(eyewin,'TextFont',el.msgfont);
```

Comment:
You can pull the TextFont and TextSize setup and restore out of the for it== loop, as the setting applies to both stereo buffers. More efficient.

Reply:
resolved in 31a44c3bca8ed8ae93777eb1b84fd8f6f12256d5